### ISSUE TITLE
Improvement to API

### ISSUE THREAD

**Issue by stuchalk** [dan2097/opsin#198]
Currently the API responds to a query that fails with an HTML page.  For programmatic access it would be better to respond with the error in json just like a successful request.

**Comment by dan2097** [comment]
I agree for the JSON endpoint, analogous JSON should still be produced for the error scenario e.g.
```
{"status":"FAILURE","message":"Cannot find in scope fragment with atom with locant 5."}
```

For the direct SMILES/CML/InChI/depiction endpoints I've changed them to return a textual error. This should be easier to understand if you're testing these APIs on say a command-line than being given a huge blob of HTML, but I don't think these APIs would be used for programmtically debugging what went wrong e.g. these APIs can't convey parsing warnings; the JSON API is more appropriate.. The text error looks like:
 
```
OPSIN has failed to process the following request:
5-amino-butane
The reason given for this failure is as follows:
Cannot find in scope fragment with atom with locant 5.
```

These changes are now live, if this isn't what you had in mind please comment on this issue.

Fixed by dan2097/opsin-ws@186cf5525e5e3b718776158945ece0621d107dd7

**Comment by stuchalk** [comment]
Thanks, that works.
